fame/core/analysis.py

```diff
@@ -200,36 +200,45 @@ def add_ioc(self, value, source, tags=[]):
                                    {'$addToSet': {'iocs.$.sources': source}})
 
     def _store_preloaded_file(self, filepath=None, fd=None):
+        if not filepath and not fd:
+            raise ValueError(
+                "Please provide either the path to the file or a file-like "
+                "object containing the data.")
+
+        if filepath and fd:
+            self.log(
+                "debug",
+                "Please provide either the path to the file or a "
+                "file-like object containing the data, not both."
+                "Choosing the filepath for now.")
+
         if fame_config.remote:
-            if not filepath and not fd:
-                raise ValueError(
-                    "Please provide either the path to the file or a file-like "
-                    "object containing the data.")
-
-            if filepath and fd:
-                self.log("debug",
-                         "Please provide either the path to the file or a "
-                         "file-like object containing the data, not both. "
-                         "Choosing the file-like object for now.")
-                response = send_file_to_remote(fd, '/files/')
-            else:
+            if filepath:
                 response = send_file_to_remote(filepath, '/files/')
+            else:
+                response = send_file_to_remote(fd, '/files/')
 
-            return File(json_util.loads(response.text)['file'])
+            return File(response.json()['file'])
         else:
-            return File(filename=os.path.basename(filepath), stream=fd)
+            if filepath:
+                with open(filepath, 'rb') as f:
+                    return File(filename=os.path.basename(filepath), stream=f)
+            else:
+                return File(filename='file', stream=fd)
```

Comment:
I'd prefer to use the submitted hash as the filename here because the filename will be used at different situations later on (eg. when downloading the file from FAME).